### PR TITLE
Photo Booth, ID Photos, and Scissors (the Database Update Update)

### PR DESCRIPTION
**Pull request by DeityLink**
I made that Photo Booth sprite and animations back in January. Sure took me a while to finally implement it. We lacked a crucial requirement at the time though, god bless getFlatIconDeluxe() !

photobooth
photo_update
backgrounds
photobooth

:cl:
* rscadd:  Added a new machine, the Photo Booth. Can be entered by just walking inside from the front, clicking it while inside will buckle you in there for a few seconds, then after a few seconds an ID photo will pop out of the machine. There will be at least one on each map, until then you can build them from R&D. You can also Alt Click the booth to select one of several backgrounds.
* rscadd: ID Photo are a new subtype of photos. Just like regular photos you can write on them, use them on newscasters, but more importantly, when using them on a Security or Medical Records computer open on a given entry, you can scan the photo on the computer to update the entry's photo. Until now, new records created at a Security Computer were damned to never have any picture, this changes at last!
* rscadd: Added scissors. They're basically just wirecutters with a different sprite. You can print them at an autolathe from the Misc Tools category. You can use them (or any type of wirecutter) on an ID photo to split it into 4 regular photos. Those individual photos CANNOT be used to update records.

### CODE REVIEW

**Comment by DrSnips** [comment]
Didn't we just add scissors in a previous PR

**Comment by DeityLink** [comment]
> 
> 
> Didn't we just add scissors in a previous PR

nah, we added barber scissors. Those shits:

image

I'm adding just a wirecutter reskin that's more akin to this both in use and purpose:

image

with big meaty grips.

**Comment by west3436** [comment]
requesting an emag feature to take an x-ray of you and show a skeleton in your clothing

**Comment by Durgen** [comment]
would tandem photos be too much work? I feel like the whole point of photobooths in the irl is to take them with qts

**Comment by DeityLink** [comment]
> 
> would tandem photos be too much work? I feel like the whole point of photobooths in the irl is to take them with qts

Well yes, this PR virtually took me "only" half the day thanks to most of the heavy lifting about player sprite capture having been done in the last photography PR (#29822) and I'd rather keep a regular stream of smaller PRs than bloat a single PR unless necessary (those who know my history of making gigantic PRs are allowed to laugh here). Anyway my point is that the main goal of this PR was giving players a way to print ID photos to update the database. The funny backgrounds were me getting side-tracked as usual because obviously I can see the potential for funny photos and doing them with several players at once would be a natural development, but since the intended goal has already been reached I'd rather submit it now that it works, and add support for multiple players in there at once (or any additional feature, emags, etc) in later PRs.

ps: as an addendum, working on several smaller PRs allows me to regularly switch which feature I'm working on which is good to stay motivated.

**Comment by Eneocho** [comment]
Add some scissors on maintenance so you can hack doors using scissors instead of wirecutters.

**Comment by DeityLink** [comment]
just ask Cargo for scissors...

File: code/game/machinery/computer/medical.dm [hunk 1]
```diff
@@ -140,6 +140,20 @@
 	onclose(user, "med_rec")
 	return
 
+
+/obj/machinery/computer/med_data/attackby(var/obj/item/O, var/mob/user)
+	if (authenticated && istype(O, /obj/item/weapon/photo/id) && (screen == 4.0) && active1)
+		var/obj/item/weapon/photo/id/photo_id = O
+		if (photo_id.four_sides)
+			if (alert("Do you want to update the records with this ID photo?",,"Yes","No") == "Yes")
+				if ((screen == 4.0) && active1)
```
Comment (Kurfursten) [inline code/game/machinery/computer/medical.dm]: probably not majorly noticeable but there's no proximity or incapacitated check after the alert meaning you can walk away and update someone's picture after dying

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -673,3 +709,162 @@
...
+	else if(mover.dir == SOUTH) //Or are we using move code
+		return 1
+	return 0
+
+/obj/machinery/photobooth/Cross(atom/movable/mover, turf/target, height = 0)
+	if(get_dir(loc, target) == SOUTH || get_dir(loc, mover) == SOUTH)
+		return 1
+	return 0
+
+/obj/machinery/photobooth/AltClick()
```
Comment (Kurfursten) [inline code/modules/paperwork/photography.dm:768]: Consider documenting this option as `info` text in `examine()`

**Comment by DeityLink** [comment]
That should be enough sanity

**Pull request merged**